### DISCUSSION TITLE
How to make it so agents can only see conversations assigned to them? (same inbox)

### DISCUSSION THREAD

**Discussion by obtusefriend** [chatwoot/chatwoot#9715] [Q&A]
Title is self-explanatory. Is it possible?
For context, I'm self-hosting Chatwoot with Typebot and made it so that conversations are assigned automatically to whomever the user wishes. I _need_ to use just one inbox.
A guy I know said I could code out the "All" and "Unassigned" lists because Chatwoot is open source. Is that really possible? If yes, I'd need to do that only for normal agents and not admin agents, which still need to see all conversations.
Thanks for any help or lead, it'll help me a lot.

**Comment by jcmatoskx** [comment]
It seems the chatwoot team is working on permissions and roles feature which should allow this to be setup.
There have been quite a few requests about this over the years, so hopefully it should be possible soon 🙏

https://github.com/chatwoot/chatwoot/issues/4216
https://github.com/orgs/chatwoot/discussions/8863
https://github.com/chatwoot/chatwoot/pull/4063

**Reply by sojan-official** [reply]
Confirming that this is being worked on at the moment

**Comment by RaphaMelnik** [comment]
Any new developments in permissions, where only the team agent can view conversations?

**Reply by jcmatoskx** [reply]
first iteration of permissions has been released on v3.13.0
https://github.com/orgs/chatwoot/discussions/10129#discussioncomment-11768580

Unfotunately it is blocked behind Enterprise / paid chatwoot plans

**Reply by RaphaMelnik** [reply]
I'm referring to when the conversation is assigned to a team, without being assigned to an agent on the team, it stays in the unassigned "box", but appears to all agents in "All Conversations" / Unassigned.
It should only appear to team agents.

**Reply by jcmatoskx** [reply]
I am not  sure about that. AFAIK only by testing in a premium/enterprise chatwoot installation

**Comment by soaadhallak** [comment]
> Title is self-explanatory. Is it possible?
> For context, I'm self-hosting Chatwoot with Typebot and made it so that conversations are assigned automatically to whomever the user wishes. I _need_ to use just one inbox.
> A guy I know said I could code out the "All" and "Unassigned" lists because Chatwoot is open source. Is that really possible? If yes, I'd need to do that only for normal agents and not admin agents, which still need to see all conversations.
> Thanks for any help or lead, it'll help me a lot.
